components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java

```diff
@@ -3442,4 +3432,80 @@
+                    + "ARSM.TENANT_ID = ? AND "
+                    + "ARSM.SCOPE_NAME NOT IN (SELECT GS.NAME FROM AM_SHARED_SCOPE GS WHERE GS.TENANT_ID = ?) AND "
+                    + "ARSM.URL_MAPPING_ID = AUM.URL_MAPPING_ID AND "
+                    + "AUM.API_ID = AM_API.API_ID";
+    public static final String IS_SCOPE_ATTACHED =
+            "SELECT 1 FROM AM_API_RESOURCE_SCOPE_MAPPING WHERE SCOPE_NAME = ? AND TENANT_ID = ?";
+
+    public static final String REMOVE_RESOURCE_SCOPE_URL_MAPPING_SQL =
+            " DELETE FROM AM_API_RESOURCE_SCOPE_MAPPING "
+                    + "WHERE URL_MAPPING_ID IN ( SELECT URL_MAPPING_ID FROM AM_API_URL_MAPPING " + "WHERE API_ID = ? )";
+
+    public static final String GET_UNVERSIONED_LOCAL_SCOPES_FOR_API_SQL =
+            "SELECT DISTINCT ARSM.SCOPE_NAME "
+                    + "FROM AM_API_RESOURCE_SCOPE_MAPPING ARSM, AM_API_URL_MAPPING AUM "
+                    + "WHERE ARSM.URL_MAPPING_ID = AUM.URL_MAPPING_ID AND "
+                    + "AUM.API_ID = ? AND ARSM.TENANT_ID = ? AND "
+                    + "ARSM.SCOPE_NAME NOT IN (SELECT GS.NAME FROM AM_SHARED_SCOPE GS WHERE GS.TENANT_ID = ?) AND "
+                    + "ARSM.SCOPE_NAME NOT IN ( "
+                    + "SELECT ARSM2.SCOPE_NAME FROM AM_API_RESOURCE_SCOPE_MAPPING ARSM2, AM_API_URL_MAPPING AUM2 "
+                    + "WHERE ARSM2.URL_MAPPING_ID = AUM2.URL_MAPPING_ID AND AUM2.API_ID != ? AND ARSM2.TENANT_ID = ?)" ;
+
+    public static final String GET_URL_TEMPLATES_WITH_SCOPES_FOR_API_SQL =
+            " SELECT AUM.URL_MAPPING_ID, "
+                    + "AUM.URL_PATTERN, "
+                    + "AUM.HTTP_METHOD, "
+                    + "AUM.AUTH_SCHEME, "
+                    + "AUM.THROTTLING_TIER, "
+                    + "AUM.MEDIATION_SCRIPT, "
+                    + "ARSM.SCOPE_NAME "
+                    + "FROM "
+                    + "AM_API_URL_MAPPING AUM "
+                    + "INNER JOIN AM_API_RESOURCE_SCOPE_MAPPING ARSM ON AUM.URL_MAPPING_ID = ARSM.URL_MAPPING_ID "
+                    + "AND AUM.API_ID = ?";
+
+    public static final String REMOVE_KM_RESOURCE_SCOPE_SQL =
+            " DELETE FROM IDN_OAUTH2_RESOURCE_SCOPE WHERE SCOPE_NAME = ? AND TENANT_ID = ? AND RESOURCE_PATH = ?";
+
+
+    public static final String GET_API_SCOPES_SQL =
+            " SELECT ARSM.SCOPE_NAME FROM AM_API_RESOURCE_SCOPE_MAPPING ARSM, AM_API_URL_MAPPING AUM "
+                    + "WHERE ARSM.URL_MAPPING_ID = AUM.URL_MAPPING_ID AND AUM.API_ID = ?";
+
+    /**
+     * Static class to hold database queries related to key management.
+     */
+    public static class KeyMgtConstants {
+
+        public static final String ADD_KM_APPLICATION =
+                "INSERT INTO AM_KM_MGT_APPLICATION (CONSUMER_KEY,CONSUMER_SECRET,TENANT_ID) VALUES (?,?,?)";
+
+        public static final String GET_KM_APPLICATION_FOR_TENANT =
+                "SELECT * FROM AM_KM_MGT_APPLICATION WHERE TENANT_ID = ?";
```

Comment:
shall we use specific fields instead of *?

Reply:
+1 Fixed with https://github.com/wso2/carbon-apimgt/pull/8409/commits/0f586a5447b97cc79c01a0ff3a4a7a94a1ce74cf